yarl/quoting.py

```diff
@@ -1,7 +1,16 @@
+import os
 import re
+import sys
 from string import ascii_letters, ascii_lowercase, digits
 from typing import Optional, TYPE_CHECKING, cast
 
+
+NO_EXTENSIONS = bool(os.environ.get("YARL_NO_EXTENSIONS"))  # type: bool
+
+if sys.implementation.name != "cpython":
```

Comment:
I prefer a higher-level API for this:
```suggestion
if platform.python_implementation() != "CPython":
```

Reply:
1. The suggestion requires additional import
2. More importantly, I would say thay `sys.implementation` is more high-level API than low-level `platform` helper module.

Reply:
Weird, I was reading somewhere that `platform` handles corner cases better...